src/core/components/courses/ProfileOrganization.vue

```diff
@@ -531,73 +527,38 @@ export default {
         this.pdfLoading = false;
       }
     },
-    async updateSalesRepresentative () {
+    async updateInterlocutor (role) {
       try {
         this.interlocutorModalLoading = true;
-        this.v$.tempInterlocutorId.$touch();
-        if (this.v$.tempInterlocutorId.$error) return NotifyWarning('Champ(s) invalide(s)');
+        this.v$.tempInterlocutor.$touch();
+        if (this.v$.tempInterlocutor.$error) return NotifyWarning('Champ(s) invalide(s)');
 
         const payload = {
-          salesRepresentative: this.tempInterlocutorId,
-          ...(get(this.course, 'contact._id') === get(this.course, 'salesRepresentative._id') &&
-            { contact: this.tempInterlocutorId }
+          [role]: this.tempInterlocutor._id,
+          ...(
+            (
+              this.tempInterlocutor.isContact ||
+              get(this.course, 'contact._id') === this.course[role]._id
```

Comment:
Pour conserver le get ici tu peux faire `get(this.course, '${role}._id')` (avec des "`")

Reply:
done